File: app/src/main/java/ru/roscha_akademii/medialib/update/UpdateModule.java
package ru.roscha_akademii.medialib.update;

import android.content.SharedPreferences;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import ru.roscha_akademii.medialib.common.ServiceNavigator;
import ru.roscha_akademii.medialib.common.TimeProvider;

@Module
public class UpdateModule {
    @Provides
    @Singleton
    public UpdateScheduler provides(SharedPreferences prefs,
                  TimeProvider timeProvider,
                  ServiceNavigator navigator) {
        return new UpdateScheduler(prefs, timeProvider, navigator);
    }
}

File: app/src/main/java/ru/roscha_akademii/medialib/update/UpdateModule.kt
package ru.roscha_akademii.medialib.update

import android.content.SharedPreferences

import javax.inject.Singleton

import dagger.Module
import dagger.Provides
import ru.roscha_akademii.medialib.common.ServiceNavigator
import ru.roscha_akademii.medialib.common.TimeProvider

@Module
open class UpdateModule {
    @Provides
    @Singleton
    open fun provides(prefs: SharedPreferences,
                 timeProvider: TimeProvider,
                 navigator: ServiceNavigator): UpdateScheduler {
        return UpdateScheduler(prefs, timeProvider, navigator)
    }
}
